pkg/diff/diff.go

```
@@ -194,6 +202,107 @@ func serverSideDiff(config, live *unstructured.Unstructured, opts ...Option) (*D
 	return buildDiffResult(predictedLiveBytes, liveBytes), nil
 }
 
+// restoreIgnoredFieldsFromLive applies normalization to detect which fields get removed, then restores them using live values.
+// This function receives original, non-normalized config and live, so it can properly detect
+// which fields are removed by normalization and restore them. This is the same pattern as normalizeTargetResources in argo-cd.
+func restoreIgnoredFieldsFromLive(config, live *unstructured.Unstructured, normalizer Normalizer) *unstructured.Unstructured {
+	if config == nil || live == nil || normalizer == nil {
+		return config
+	}
+
+	// Apply normalization to config to see what fields get removed (same as normalizeTargetResources)
+	originalConfig := config.DeepCopy()
+	normalizedConfig := config.DeepCopy()
+	err := normalizer.Normalize(normalizedConfig)
+	if err != nil {
+		// If normalization fails, return original config
+		return originalConfig
+	}
+
+	// Calculate what fields were removed from live by normalization (same pattern as livePatch in normalizeTargetResources)
+	var lookupPatchMeta *strategicpatch.PatchMetaFromStruct
+	versionedObject, err := scheme.Scheme.New(originalConfig.GroupVersionKind())
+	if err == nil {
+		meta, err := strategicpatch.NewPatchMetaFromStruct(versionedObject)
+		if err == nil {
+			lookupPatchMeta = &meta
+		}
+	}
```

what is the reason why we are ignoring the errors in this case?

---

In this case if there is an error getMergePatch() will fallback to JSON merging as lookupPatchMeta will be nil.
But yes we don't want to ignore it here, we should at least log it.